src/System.Collections.Immutable/src/System.Collections.Immutable.csproj

```diff
@@ -40,6 +40,7 @@
     <Compile Include="System\Collections\Immutable\ImmutableArray_1.Builder.cs" />
     <Compile Include="System\Collections\Immutable\ImmutableArray_1.Enumerator.cs" />
     <Compile Include="System\Collections\Immutable\ImmutableArray_1.cs" />
+    <Compile Include="System\Collections\Immutable\ImmutableArray_1.FullApi.cs" />
```

Comment:
The idea here, which it seems is unclear, is that a code consumer would either compile ImmutableArray_1.cs and ImmutableArray_1.FullApi.cs, or ImmutableArray_1.cs and ImmutableArray_1.PartialApi.cs. Its a technique for avoiding #ifs (which is considered important in some of the codebases I work in.) The other technique is to put a bunch of #ifs in the code.

All of this is around allowing ImmutableArray's source code to be used by a component which due to layering concerns must execute without access to higher levels apis that happen to be used in the FullApi.cs implementation. (In particular, there is a need for SortedDictionary which is considered too high level for my uses.)